CMakeLists.txt

```diff
@@ -438,7 +448,7 @@ endif()
 
 set(USE_X11 0)
 
-if(UNIX AND NOT APPLE AND NOT ANDROID AND NOT ENABLE_HEADLESS)
+if(UNIX AND NOT APPLE AND NOT ANDROID AND NOT HAIKU AND NOT ENABLE_HEADLESS)
```

Comment:
Haiku has its own display server/window manager which is not even remotely compatible with X11. And if Dolphin can't find X11, you run into lines 465 and following:
```
if (NOT USE_X11)
 		message(FATAL_ERROR "\n"
 			"No suitable display platform found\n"
 			"Requires x11 to run")
endif()
```
So Haiku is basically equivalent to macOS/Android in this regard, and they're also special cased here.